crates/query-engine/metadata/src/metadata/native_queries.rs

```diff
@@ -120,9 +240,18 @@ impl JsonSchema for NativeQuerySql {
 
 // Parsing
 
-/// Parse a native query into parts where variables have the
-/// syntax `{{<variable>}}`.
-fn parse_native_query(string: &str) -> Vec<NativeQueryPart> {
+/// Read a file a parse it into native query parts.
+pub fn parse_native_query_from_file(file: std::path::PathBuf) -> Result<NativeQuerySql, String> {
+    let contents: String = match fs::read_to_string(&file) {
+        Ok(ok) => Ok(ok),
+        Err(err) => Err(format!("{}: {}", &file.display(), err)),
```

Comment:
Again, here seems like a good place for `Error::IoError { path: Option<PathBuf>, inner: io::Error }`.

Reply:
I would approve such a PR :)